docs/IR.md

```diff
@@ -370,7 +370,9 @@ Which is referenced by the Tensor type message:
   }
 ```
 
-The empty list of dimension sizes, [], is a valid tensor shape, denoting a zero-dimension (scalar) value. A zero-dimension tensor is distinct from a tensor of unknown dimensionality, which is indicated by an absent 'shape' property in the Tensor record. When the shape property is absent for an input, a tensor value of any shape may be passed from the caller. When the shape property is absent for an output, the caller should expect that the output value may be of any shape.
+The empty list of dimension sizes, [], is a valid tensor shape, denoting a zero-dimension (scalar) value. A zero-dimension tensor is distinct from a tensor of unknown dimensionality, which is indicated by an absent 'shape' property in the Tensor record. When the shape property is absent for a node input, a tensor value of any shape may be passed from the caller. When the shape property is absent for a node output, the caller should expect that the output value may be of any shape. 
+
+Note that node input and output can be completely absent from the graph (and can be filled by shape inference), but graph input and output must always specify a shape.
```

Comment:
I suggested a reworded version which merges this statement into the previous paragraph. If you want the first part of the above, I suggest using "Note that type and shape information for intermediate values in the graph, used as inputs and outputs of nodes, are optional but not compulsory as they can be inferred, but type and rank information is compulsory for inputs and outputs of the model.